### DISCUSSION TITLE
Allow .html path and possibly redirect

### DISCUSSION THREAD

**Discussion by kickbk** [gatsbyjs/gatsby#38079] [Help] [answered]
I am migrating an existing website to Gatsby.
Currently allow that website paths end with ".html", like `https://mywebsite.com/invited.html?i=me&y=you`

In the example I write above, I parse the url params and use them in to customize the page.

I would like to allow both `https://mywebsite.com/invited.html?i=me&y=you` and  `https://mywebsite.com/invited?i=me&y=you` or redirect to `https://mywebsite.com/invited?i=me&y=you`

How do I do that?

**Comment by gh640** [comment, marked as answer]
If you need to make redirect settings, you usually need to do that on your server. How to do that depends on the server / service you use. `createRedirect()` might help:

https://www.gatsbyjs.com/docs/how-to/cloud/working-with-redirects-and-rewrites/

It's not recommended to provide the same content with different urls. But if you really want, you need to call `createPage()` for those urls like:

```js
createPage({
  path: `/invited`,
  component,
  context: {
    ...,
  },
})

createPage({
  path: `/invited.html`,
  component,
  context: {
    ...,
  },
})
```

More detailed explanation is in the following page:

https://www.gatsbyjs.com/docs/creating-and-modifying-pages/
